Week3/homework/step3-bonus.js

```diff
@@ -1,14 +1,44 @@
 'use strict';
 
 const values = ['a', 'b', 'c', 'd', 'a', 'e', 'f', 'c'];
-
+const values2 = ['s', 'f', 'g', 's', 'r', 't', 't', 'a', 'f'];
+const values3 = [
+  '22',
+  '33',
+  '2',
+  '4',
+  '54',
+  '33',
+  '67',
+  '76',
+  '77',
+  '66',
+  '77',
+  '67',
+  '8',
+  '4',
+  '2',
+  '12',
+  '11',
+  '54',
+];
 function makeUnique(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = i + 1; j < arr.length; j++) {
+      if (arr[i] === arr[j]) {
+        arr.splice(j, 1);
```

Comment:
While this works, I would recommend against deleting elements from an array while you are relying on its (changing) length. It is like trying to remove the leg of a chair you sit on. I would prefer a method that creates a new array with unique values without modifying (i.e., 'mutating') the existing array. Can you think of such a method?

Reply:
I updated my code using a new array and without touching source array.